pkg/apis/volumesnapshot/v1beta1/types.go

```diff
@@ -0,0 +1,354 @@
+
+	// volumeSnapshotContentName specifies the name of a pre-existing VolumeSnapshotContent
+	// object a user asks to statically bind the VolumeSnapshot object to.
+	// +optional
+	VolumeSnapshotContentName *string `json:"volumeSnapshotContentName,omitempty" protobuf:"bytes,2,opt,name=volumeSnapshotContentName"`
+}
+
+// VolumeSnapshotStatus is the status of the VolumeSnapshot
+type VolumeSnapshotStatus struct {
+	// NOTE: All fields in VolumeSnapshotStatus are informational for user references.
+	// Controllers MUST NOT rely on any fields programmatically.
+
+	// boundVolumeSnapshotContentName represents the name of the VolumeSnapshotContent
+	// object to which the VolumeSnapshot object is bound.
+	// If not specified, it indicates that the VolumeSnapshot object has not been
+	// successfully bound to a VolumeSnapshotContent object yet.
+	// +optional
+	BoundVolumeSnapshotContentName *string `json:"boundVolumeSnapshotContentName,omitempty" protobuf:"bytes,1,opt,name=boundVolumeSnapshotContentName"`
+
+	// creationTime, if not nil, represents the timestamp when the point-in-time
+	// snapshot was successfully cut on the underlying storage system.
+	// In dynamic snapshot creation case, it will be filled in upon snapshot creation.
+	// For a pre-existing snapshot, it will be filled in once the VolumeSnapshot object has
+	// been successfully bound to a VolumeSnapshotContent object and the underlying
+	// storage system has the information available.
+	// If not specified, it indicates that the creation time of the snapshot is unknown.
+	// +optional
+	CreationTime *metav1.Time `json:"creationTime,omitempty" protobuf:"bytes,2,opt,name=creationTime"`
+
+	// readyToUse indicates if a snapshot is ready to be used to restore a volume.
+	// In dynamic snapshot creation case, readyToUse will be set to true after underlying storage
+	// system has successfully finished all out-of-bound procedures to make a snapshot ready to
+	// be used to restore a volume.
+	// For a pre-existing snapshot, readyToUse will be set to the value returned
+	// from CSI "ListSnapshots" gRPC call if the matching CSI driver exists and supports.
+	// Otherwise, if there exists no CSI driver or the driver does not support "ListSnapshots",
+	// this field will be set to "True".
+	// If not specified, it indicates that the readiness of a snapshot is unknown.
+	// +optional
+	ReadyToUse *bool `json:"readyToUse,omitempty" protobuf:"varint,3,opt,name=readyToUse"`
+
+	// restoreSize represents the complete size of the snapshot in bytes.
+	// The purpose of this field is to give user guidance on how much space is
+	// needed to restore a volume from this snapshot.
+	// When restoring a volume from a snapshot, the size of the volume MUST NOT
+	// be less than the restoreSize. Otherwise the restoration will fail.
+	// If this field is not specified, it indicates that underlying storage system
```

Comment:
Can you get it from ListSnapshots?

Reply:
potentially yes, however it's not a required gRPC call for CSI driver. It's possible the underlying storage system does not support it.

Reply:
For some fields, we say that we'll get the information from ListSnapshots if the driver supports it, and other fields we don't say that. Can we make the comments consistent across all the fields?